src/Ionide.ProjInfo/Library.fs

```diff
@@ -91,29 +200,29 @@ module ProjectLoader =
                    "TargetFramework", tfm.Value
                if path.EndsWith ".csproj" then
                    "NonExistentFile", Path.Combine("__NonExistentSubDir__", "__NonExistentFile__")
-               "DotnetProjInfo", "true" 
+               "DotnetProjInfo", "true"
+               "MSBuildEnableWorkloadResolver", "false" // we don't care about workloads
                yield! globalProperties ]
 
 
     let buildArgs =
-        [| "ResolvePackageDependenciesDesignTime"
+        [| "ResolveAssemblyReferencesDesignTime"
+           "ResolveProjectReferencesDesignTime"
+           "ResolvePackageDependenciesDesignTime"
            "_GenerateCompileDependencyCache"
+           "_ComputeNonExistentFileProperty"
            "CoreCompile" |]
 
     let loadProject (path: string) (generateBinlog: bool) (ToolsPath toolsPath) globalProperties =
         try
-            let tfm = getTfm path
-
-            let globalProperties = getGlobalProps path tfm globalProperties
+            let readingProps = getGlobalProps path None toolsPath globalProperties
+            let tfm = getTfm path readingProps
 
-            match System.Environment.GetEnvironmentVariable "DOTNET_HOST_PATH" with
-            | null
-            | "" -> System.Environment.SetEnvironmentVariable("DOTNET_HOST_PATH", Ionide.ProjInfo.Paths.dotnetRoot)
-            | _alreadySet -> ()
+            let globalProperties = getGlobalProps path tfm toolsPath globalProperties
```

Comment:
we have to resolve props twice here: once to accurately get the TFM resolved, once to get the 'full' tfm-included props resolved.